File: Principal/Html/Pagina/PagPrincipal.cs
using NetZ.Web;
using NetZ.Web.DataBase.Dominio;
using NetZ.Web.Html;
using NetZ.Web.Html.Pagina;
using NetZ.Web.Server.Arquivo.Css;
using Sentinela.Html.Componente.Activity;
using Web.UiManager;

namespace Sentinela.Html.Pagina
{
    [HtmlExport(true)]
    internal sealed class PagPrincipal : PagMobile
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        #endregion Atributos

        #region Construtores

        public PagPrincipal() : base(AppSentinela.i.strNome)
        {
        }

        #endregion Construtores

        #region Métodos

        protected override void addJs(LstTag<JavaScriptTag> lstJs)
        {
            base.addJs(lstJs);

            lstJs.Add(new JavaScriptTag(AppWebBase.DIR_JS + "sentinela/dominio/ServidorDominio.js", 201));
            lstJs.Add(new JavaScriptTag(typeof(AppSentinela)));
            lstJs.Add(new JavaScriptTag(typeof(DominioWebBase)));
        }

        protected override void addLayoutFixo(JavaScriptTag tagJs)
        {
            base.addLayoutFixo(tagJs);

            tagJs.addLayoutFixo(typeof(ActServidorCadastro));
            tagJs.addLayoutFixo(typeof(ActServidorDetalhe));
        }

        protected override bool getBooJsAutoInicializavel()
        {
            return true;
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            new ActSplashScreen().setPai(this);

            new ActHome().setPai(this);
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            this.addCss(css.setBottom(0));
            this.addCss(css.setLeft(0));
            this.addCss(css.setPosition("fixed"));
            this.addCss(css.setRight(0));
            this.addCss(css.setTop(0));
            this.addCss(css.setColor(AppSentinela.i.objTema.corFonte));
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: Sentinela.TypeScript/html/pagina/PagPrincipal.ts
/// <reference path="../../../Web.TypeScript/html/pagina/PagMobile.ts"/>
/// <reference path="../../../Web.TypeScript/Utils.ts"/>

module Sentinela
{
    // #region Importações

    import PagMobile = Web.PagMobile;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class PagPrincipal extends PagMobile
    {
        // #region Constantes

        // #endregion Constantes

        // #region Atributos

        private static _i: PagPrincipal;

        public static get i(): PagPrincipal
        {
            if (PagPrincipal._i != null)
            {
                return PagPrincipal._i;
            }

            PagPrincipal._i = new PagPrincipal();

            return PagPrincipal._i;
        }

        private _actHome: ActHome;
        private _actSplashScreen: ActSplashScreen;

        private get actHome(): ActHome
        {
            if (this._actHome != null)
            {
                return this._actHome;
            }

            this._actHome = new ActHome();

            return this._actHome;
        }

        private get actSplashScreen(): ActSplashScreen
        {
            if (this._actSplashScreen != null)
            {
                return this._actSplashScreen;
            }

            this._actSplashScreen = new ActSplashScreen();

            return this._actSplashScreen;
        }

        // #endregion Atributos

        // #region Construtores
        // #endregion Construtores

        // #region Métodos

        public abrirActServidorCadastro(): void
        {
            this.actHome.esconder();

            var actServidorCadastro = new ActServidorCadastro();

            this.tagBody.addHtml(actServidorCadastro.strLayoutFixo);

            actServidorCadastro.iniciar();
        }

        public fecharSplash(): void
        {
            this.actHome.booVisivel = true;

            this.actSplashScreen.esconder(undefined, (() => this.actSplashScreen.dispose()));
        }

        protected inicializar(): void
        {
            super.inicializar();

            this.inicializarApp();

            this.actHome.iniciar();

            this.actSplashScreen.iniciar();
        }

        private inicializarApp(): void
        {
            AppSentinela.i.pag = this;

            AppSentinela.i.iniciar();
        }

        public salvar(objServidor: ServidorDominio): void
        {
            AppSentinela.i.salvar(objServidor);

            this.actHome.mostrar();
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}